app/scripts/controllers/build.js

```diff
@@ -114,24 +117,43 @@ angular.module('openshiftConsole')
         );
 
         watches.push(DataService.watch("builds", context, function(builds, action, build) {
-          $scope.builds = {};
-          // TODO we should send the ?labelSelector=buildconfig=<name> on the API request
-          // to only load the buildconfig's builds, but this requires some DataService changes
-          var allBuilds = builds.by("metadata.name");
-          angular.forEach(allBuilds, function(build, name) {
-            if (build.metadata.labels && build.metadata.labels.buildconfig === $routeParams.buildconfig) {
-              $scope.builds[name] = build;
+          if (!action) {
```

Comment:
Curious, could ~ 120-139 be put on a buildConfig service and reused in  buildConfig.js?  Thinking `$scope.builds = buildConfigService.validateBuilds(builds, action, build);` might be able to eliminate duplication.